test/ed25519-test.js

```diff
@@ -0,0 +1,138 @@
+/* -------------------------------- REQUIRES -------------------------------- */
+
+var assert = require('assert');
+var fs = require('fs');
+var bn = require('bn.js');
+var hash = require('hash.js');
+var elliptic = require('../');
+var utils = elliptic.utils;
+var toArray = elliptic.utils.toArray;
+var eddsa = elliptic.eddsa;
+
+/* --------------------------------- HELPERS -------------------------------- */
+
+function toHex(arr) {
+  return elliptic.utils.toHex(arr).toUpperCase();
+}
+
+/* ---------------------------------- TESTS --------------------------------- */
+
+var MAX_PROGRAMMATIC = process.env.CI ? Infinity : 50;
+
+describe('ed25519 derivations', function() {
+  var expectedTests = 256;
+  var ed25519, derivations;
+
+  before(function() {
+    ed25519 = new eddsa('ed25519');
+    derivations = require('./fixtures/derivation-fixtures');
+    assert.equal(derivations.length, expectedTests);
+  });
+
+  function testFactory(i) {
+    it('can compute correct a and A for secret: ' + i, function() {
+      var test = derivations[i];
+      var secret = utils.toArray(test.secret_hex, 'hex');
+      var key = ed25519.keyFromSecret(secret);
+      assert.equal(toHex(key.privBytes()), test.a_hex);
+      var xRecovered = toHex(ed25519.encodeInt(
+                              ed25519.decodePoint(key.pubBytes()).getX()));
```

Comment:
Btw, why so much magic? Isn't it just possible to do `.getX().toString('hex', 2)`?